akka-javasdk/src/test/java/akka/javasdk/agent/DummyAgent.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.javasdk.agent;
+
+class DummyAgent1 extends ChatAgent {
+  Effect<String> doSomething(String question) {
+    return effects()
+        .systemMessage("You are a helpful...")
+        .userMessage(question)
+        .thenReply();
+  }
+}
+
+class DummyAgent2 extends ChatAgent {
+  record Response(String result) {}
+
+  Effect<Response> doSomething(String question) {
```

Comment:
We had a discussion about that an agent can only have one such method, and one suggestion was to make it a predefined abstract method. One drawback of that would be that we can't have a different effect type to indicate streaming responses, as we have for views.

Reply:
If we have more than one method, then we open the door to have agents with different prompts. 

Actually, we are already doing it. 

```java
if (someCondition) return effect().systemMessage("this prompt")...
else return effect().systemMessage("this other prompt")...
```

Not sure if we will be promoting an anti-pattern or leaving the door open for it. 

Alternatively, the `ChatAgent` could have a method that returns the system prompt and as such we enforce a single prompt all the all methods in the agent. 

Anyway, maybe I'm over-thinking it.

Reply:
I will add a runtime validation that it can only have one, and only one, such command handler. We could open up for more methods later if we see a motivation for it (e.g. same agent supporting both streaming and strict response).